src/committee/orchestrator.rs

```diff
@@ -268,24 +269,30 @@ pub struct Orchestrator {
     pub pubkey_package: frost_secp256k1_tr::keys::PublicKeyPackage,
     pub committee_cfg: CommitteeConfig,
     pub member_status: Arc<RwLock<MemberStatusState>>,
+    compliance: Arc<Compliance>,
 }
 
 impl Orchestrator {
     pub fn new(
         pubkey_package: frost_secp256k1_tr::keys::PublicKeyPackage,
         committee_cfg: CommitteeConfig,
         member_status: Arc<RwLock<MemberStatusState>>,
+        compliance: Arc<Compliance>,
     ) -> Self {
         Self {
             pubkey_package,
             committee_cfg,
             member_status,
+            compliance,
         }
     }
 
     /// Handles bob request from A to Z.
     pub async fn handle_request(&self, bob_request: &BobRequest) -> Result<BobResponse> {
         // Validate transaction before forwarding it, and get smart contract
+        bob_request
+            .check_compliance(Arc::clone(&self.compliance))
+            .await?;
```

Comment:
wondering: should individual nodes perform that check as well? What would be the rational for not performing that check at the node level (and would node be liable if something bad happens there...)

Reply:
We could, but we've got orchestrator as a centralized entity that coordinates the entire process.